### ISSUE TITLE
NgFor with NgIf does not produce the correct results (Beta1)

### ISSUE THREAD

**Issue by tine2k** [angular/angular#6304]
I am using NgFor on a `<template>` tag to render list items. Every iteration renders a list item. In addtion, every other iteration should render another list item (for every even index number, conditional with NgIf).

Have a look at this: http://codepen.io/tine2k/pen/ZQLZav?editors=001
Reproduce the bug by hitting the button several times and watch the wrong list item order. It should be "0, even number, 1, 2, even number, 3, 4, even number" and so on.

This works fine by hiding the list item using ngClass: http://codepen.io/tine2k/pen/bEqxBN?editors=001

**Comment by pe8ter** [comment]
What's really strange is that putting a space between the closing tags `</li></template>` on line 8 fixes the problem in http://codepen.io/tine2k/pen/ZQLZav?editors=001

**Comment by tine2k** [comment]
Dude, what a fix! :)

**Comment by tine2k** [comment]
Anyway, this still seems to be a real issue that has to be addressed.

**Comment by pe8ter** [comment]
Definitely!

**Comment by gsans** [comment]
Legendary way to approach bugs! Congrats =)

**Comment by pe8ter** [comment]
This one is even more janked in Beta.1: http://plnkr.co/edit/ZWiQqw3nR8xS1MRrpTry?p=info

**Comment by ericmartinezr** [comment]
@pe8ter you can always use `even` or `odd` properties

```
<template ngFor #item [ngForOf]="items" #i="index" #isEven="even" #isOdd="odd">
   <li>{{i}}</li>
   <li>{{isEven ? 'This number is even' : 'This number is not even'}}</li>
   <li>{{isOdd ? 'This number is odd' : 'This number is not odd'}}</li>
</template>
```
This uses the internal NgFor properties and avoid the error message. But it is still weird that awful message when using some condition inside the `<template>` (probably it is related to nested `<template>`?).

**Comment by pe8ter** [comment]
Are nested templates off limits?

**Comment by tine2k** [comment]
@ericmartinezr thanks for your help. In my real world example the if condition is more complex than even/odd. I simplified it to show the rendering bug.

**Comment by tine2k** [comment]
@pe8ter see #6380

**Comment by tine2k** [comment]
Updated my codepen example to Beta1. Still broken.

**Comment by tine2k** [comment]
Updated my codepen example to Beta2. Still broken.

**Comment by tine2k** [comment]
Updated my codepen example to Beta3. Still broken.

**Comment by tbosch** [comment]
Thanks for reporting!

**Comment by tine2k** [comment]
Updated my codepen example to Beta6. Still broken.

**Comment by warner-pinz** [comment]
How to use *ngIf  with odd and even?

**Comment by zoechi** [comment]
Have you tried
@warner-pinz https://github.com/angular/angular/issues/6304#issuecomment-170201445
with
`*ngIf="odd"

**Comment by warner-pinz** [comment]
@zoechi Thanks for that. It's working now =)

**Comment by tine2k** [comment]
Confirmed fixed in 2.0.0.Beta8. Thanks for the fix @tbosch !
